### ISSUE TITLE
Hide blocks from inserter when `multiple` is set to false

### ISSUE THREAD

**Issue by roykho** [WordPress/gutenberg#61415]
## What problem does this address?

Currently we have a definition in the `block.json` to say `supports: { "multiple": false }`. This in theory should prevent a block to appear or to be added more than once. However this doesn't actually do that. Instead it just displays a message stating `'This block can only be used once.'` but in fact it will still appear in the editor and in the frontend.

## What is your proposed solution?

I think it would be more intuitive if the block is hidden from the inserter once there is already a block present. In that way, you can't even add another block to the page.

**Comment by richtabor** [comment]
It's tough auto-hiding UI, especially with search results, as you'd question why it's not rendering. Alternatively perhaps the block should be disabled within the inserter, or a snackbar message is rendered indicating it can not be added multiple times—instead of adding a block notice in the editor.

**Comment by roykho** [comment]
Yes a snackbar message and shown as disabled in the search/inserter would be a good alternative approach. Thanks for suggesting that.

**Comment by Mamaduka** [comment]
Similar blocks already appear as disabled after inserting once. I'm going to close this in favor of #40434, which also suggests making disabling reason clear.
